src/coreComponents/physicsSolvers/contact/SolidMechanicsEmbeddedFractures.cpp

```diff
@@ -664,18 +664,19 @@ void SolidMechanicsEmbeddedFractures::applySystemSolution( DofManager const & do
     updateJump( dofManager, domain );
   }
 
-
+  std::vector< NeighborCommunicator::SyncFieldsID > fieldsTobeSync;
```

Comment:
I think that you should change this to
```c++
  std::vector< NeighborCommunicator::SyncFieldsID > fieldsTobeSync;
  forMeshTargets( domain.getMeshBodies(), [&] ( string const &,
                                                MeshLevel & mesh,
                                                arrayView1d< string const > const & )
  {
    NeighborCommunicator::SyncFieldsID fieldsID( getFractureRegionName(), NeighborCommunicator::Location::Elem );
    fieldsID.addField( string( extrinsicMeshData::contact::dispJump::key() ) );
    fieldsID.addField( string( extrinsicMeshData::contact::deltaDispJump::key()) );
    fieldsTobeSync.emplace_back( fieldsID );
  } );
  CommunicationTools::getInstance().synchronizeFields2( fieldsTobeSync,
                                                        mesh,
                                                        domain.getNeighbors(),
                                                        true );
```
or at least (I think it's less good)
```c++
  forMeshTargets( domain.getMeshBodies(), [&] ( string const &,
                                                MeshLevel & mesh,
                                                arrayView1d< string const > const & )
  {
    std::vector< NeighborCommunicator::SyncFieldsID > fieldsTobeSync;
    NeighborCommunicator::SyncFieldsID fieldsID( getFractureRegionName(), NeighborCommunicator::Location::Elem );
    fieldsID.addField( string( extrinsicMeshData::contact::dispJump::key() ) );
    fieldsID.addField( string( extrinsicMeshData::contact::deltaDispJump::key()) );
    fieldsTobeSync.emplace_back( fieldsID );
    CommunicationTools::getInstance().synchronizeFields2( fieldsTobeSync,
                                                          mesh,
                                                          domain.getNeighbors(),
                                                          true );
  } );
```
Because your implementation feeds `fieldsTobeSync` at each step of the loop and `synchronizes` fields that have already been synchronized because `fieldsTobeSync` is never flushed.

On another subject (it's a bit of a detail), having a `SyncFieldsID( string const regionName, Location const location, std::vector< string > & fieldNames)` could let you `fieldsTobeSync.emplace_back( getFractureRegionName(), NeighborCommunicator::Location::Elem, { extrinsicMeshData::contact::dispJump::key(), extrinsicMeshData::contact::deltaDispJump::key() } )`.

Also, for `fieldsID.addField( string( extrinsicMeshData::contact::dispJump::key() ) );`, maybe can you implement the proper `fieldsID.addField` such that you do not need the explicit `string( ... )` in the call?

Reply:
Unfortunately, the first one won't work because we need to pass the `mesh`  and that's what we are looping on so the `synchronizeFields` call has to be inside the lambda.